obspy/io/hypodd/pha.py

```diff
@@ -124,6 +124,56 @@ def _read_pha(filename, inventory=None, id_map=None, id_default='.{}..{}',
     return Catalog(events)
 
 
+PHA1 = ('#  {o.time.year}    {o.time.month}   {o.time.day}    {o.time.hour}   '
+        '{o.time.minute}   {o.time.second}.{o.time.microsecond:06d}   '
+        '{o.latitude}   {o.longitude}   {depth}   {mag}  0.0   0.0   {rms}'
+        '       {evid}\n')
+PHA2 = '{p.waveform_id.station_code}  {reltime:.4f}  1.0  {p.phase_hint}\n'
+
+
+def _write_pha(catalog, filename,
+               **kwargs):  # @UnusedVariable,
+    """
+    Write a HypoDD PHA file.
+
+    .. warning::
+        This function should NOT be called directly, it registers via the
+        the :meth:`~obspy.core.event.Catalog.write` method of an
+        ObsPy :class:`~obspy.core.event.Catalog` object, call this instead.
+
+    :type catalog: :class:`~obspy.core.event.catalog.Catalog`
+    :param catalog: The ObsPy Catalog object to write.
+    :type filename: str or file
+    :param filename: Filename to write or open file-like object.
+    """
+    lines = []
+    for event in catalog:
+        try:
+            ori = event.preferred_origin() or event.origins[0]
+        except IndexError:
+            continue
+        try:
```

Comment:
It might be a good idea to allow for events without magnitudes - hypoDD does not need magnitudes to function.

Reply:
Yes, see todo list.